src/inmanta/ast/statements/call.py

```diff
@@ -263,7 +263,6 @@ class FunctionUnit(Waiter):
     def __init__(self, queue_scheduler, resolver, result: ResultVariable, requires, function: FunctionCall) -> None:
         Waiter.__init__(self, queue_scheduler)
         self.result = result
-        result.set_provider(self)
         # requires is used to track all required RV's
```

Comment:
shouldn't this also become a promisse?

Reply:
In practice for each of your comments we're dealing with plain `ResultVariable`, which don't do anything with the promises they get (`self.provider` was only used for `OptionVariable`). In addition these variables get passed around to methods that expect a `ResultVariable` rather than an `IPromise`.

Reply:
Ok, I see, we may have to document this detail (that even if it has the capability to hand out promises, it is not used and not usefull?)